modules/rational-modeline.el

```diff
@@ -0,0 +1,134 @@
+;;;; rational-modeline.el --- rational module for setting up the modeline  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2022
+;; SPDX-License-Identifier: MIT
+
+;; Authors: System Crafters Community, David "System Crafters" Wilson, Erik Lundstedt
+
+;; Commentary
+
+;;
+
+;;; Code:
+
+(defcustom rational-modeline/type "doom"
+  "one of 'GNU' 'doom' 'telephone' ";;'space WIP' "
+  )
+
+
+(when (string-equal rational-modeline/type "GNU")
```

Comment:
Each `when` would need to be handled when the custom variable changes.

Reply:
How would I do this?

I am pretty new to elisp and would be glad for any help you can give

Reply:
Check out the implementation of `rational-ui-display-line-numbers` in rational-ui.el. Notice the `:set` on the custom variable and how it responds to the variable being changed. Just make sure you use `set-default` in the body of your `:set` to ensure the value actually gets changed. You can test by setting the value (with `custom-set-variables`). Try setting the value both before and after `require`ing `rational-modeline` to see the effect.  Happy to help if you have questions after looking at `rational-ui-display-line-numbers`.